src/DynamoCore/Nodes/VariableInputNode.cs

```diff
@@ -73,13 +73,13 @@ public override int GetInputIndexFromModel()
         protected virtual void RemoveInput()
         {
             VariableInputController.RemoveInputBase();
-            OnAstUpdated();
+            //OnAstUpdated();
```

Comment:
Yeah this is taken care of by the `MarkNodeDirty` call in `RemoveInputFromModel` and `AddInputToModel`